app/models/state_changers.rb

```diff
@@ -168,16 +176,39 @@ def labware
     end
   end
 
-  # The tube state changer is used by Tubes. It works the same way as the default
-  # plate state changer but does not need to handle a subset of wells like the plate.
-  class TubeStateChanger < BaseStateChanger
-    # Tubes have no wells so contents is always empty
-    def contents_for(_target_state)
-      nil
+  # This version of the AutomaticLabwareStateChanger is used by TubeRacks.
+  class AutomaticTubeRackStateChanger < BaseTubeRackStateChanger
+    include AutomaticBehaviour
+
+    def labware
+      @labware ||= Sequencescape::Api::V2.tube_rack_for_completion(labware_uuid)
+    end
+
+    def move_to!(state, reason = nil, customer_accepts_responsibility = nil)
+      super
+      complete_outstanding_requests
```

Comment:
Are the requests to be closed on the tube rack or the tubes?
Do we need a custom method to loop through the tubes of the tuberack?